assets/js/projectComponents/pages/components/PageForm.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+    <v-form @submit.prevent="createPage">
+        <v-text-field
+            required
+            label="Name"
+            :counter="80"
+            v-model="page.name"
+            :error-messages="nameErrors"
+            @input="fieldChanged('name')"
+            @blur="$v.page.name.$touch()"
+            ref="nameField"
+        ></v-text-field>
+        <markdown-editor :content="initDataContent" @input="updateFormContent"></markdown-editor>
+        <v-btn depressed @click="createPage">Submit</v-btn>
+    </v-form>
+</template>
+
+<script>
+import MarkdownEditor from './MarkdownEditor'
+import { validationMixin } from 'vuelidate'
+import { maxLength, minLength, required } from 'vuelidate/lib/validators'
+
+export default {
+    name: "PageForm",
+    components: { MarkdownEditor },
+    mixins: [validationMixin],
+    validations: {
+        page: {
+            name: { required, maxLength: maxLength(80), minLength: minLength(3) }
+        }
+    },
+    props: {
+        initData: {
+            type: Object,
+            default() {
+                return {}
+            }
+        }
+    },
+    data() {
+        return {
+            page: {
+                name: '',
+                content: ''
+            }
+        }
+    },
+    mounted() {
+        this.$refs.nameField.focus()
+        if (this.initData.hasOwnProperty('name')) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#using_hasownproperty_as_a_property_name

`hasOwnProperty` should be used on `Object` instead of on object instance as this can be a dangerous.
`Object.prototype.hasOwnProperty.call(foo, 'bar');` or we can use `has` lib https://www.npmjs.com/package/has

Reply:
that's good to know, In the future we may switch to `Object.hasOwn` as it's already supported by the major browsers. I'll switch to `Object.prototype.hasOwnProperty.call` for now